PyInstaller/utils/win32/winmanifest.py

```diff
@@ -1062,7 +1070,20 @@ def create_manifest(filename, manifest, console, uac_admin=False, uac_uiaccess=F
         manifest.requestedExecutionLevel = 'requireAdministrator'
     if uac_uiaccess:
         manifest.uiAccess = True
-    manifest.writeprettyxml(filename)
+
+    # only write a new manifest if it is different from the old
+    filename_new = filename + '.new'
+    manifest.writeprettyxml(filename_new)
```

Comment:
Please *read* the old file instead of writing the new one unconditionally. This uses less disk access (and filecmp`s cache is of now use since pyinstaller typically comes here only once per run).